client/src/components/CommentForm.jsx

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { createNewComment } from "../managers/comment";
+
+export const CommentForm = ({ loggedInUser }) => {
+  const [commentObject, setCommentObject] = useState({});
+  const { postId } = useParams();
+  const navigate = useNavigate();
+
+  const handleSave = (e) => {
+    e.preventDefault();
+
+    const newComment = {
+      Subject: commentObject.subject,
+      Content: commentObject.content,
+      PostId: parseInt(postId),
+      CommenterId: loggedInUser.id,
+    };
+
+    createNewComment(newComment).then(() => {
+      navigate(`/post/${postId}`);
+    });
+  };
+
+  return (
+    <main>
+      <div>
+        <form onSubmit={handleSave}>
+          <input
+            required
+            defaultValue={commentObject.subject}
```

Comment:
Is there a reason this is `defaultValue` and not `value`? Seems like this should be a controlled element.

Reply:
It's considered an uncontrolled element when using value, probably due to subject not existing till you input something into the onchange.

Reply:
It is considered a controlled element when using value. I assume you're talking about the error you get when you change the field. Initially, value will be undefined as `commentObject.subject` is undefined. When value is undefined, react treats it as uncontrolled. When it becomes defined, it switches to being controlled, which react doesn't like for some reason. You can resolve this error by setting an initial value for `commentObject.subject` (probably an empty string).